src/dns.mli

```diff
@@ -791,6 +791,49 @@ module Edns : sig
       buffer. *)
 end
 
+(** Loc
+
+    A locator record (LOC) is used to express location information associated with domain.
+    
+    See RFC 1876. *)
+module Loc : sig
+  type t = {
+    latitude : int32;
+    longitude : int32;
+    altitude : int32;
+    size : int;
+    horiz_pre : int;
+    vert_pre : int;
+  }
+  (** The type of a Loc record. *)
+
+  val parse :
+    latitude:((int32 * int32 * int32) * [ `North | `South ])
+    -> longitude:((int32 * int32 * int32) * [ `East | `West ])
+    -> altitude:int64
+    -> precision:(int64 * int64 * int64)
```

Comment:
it is likely perfectly fine, but why is precision here a triple of int64, above in the type a triple of int? is this fine for 32 bit systems?

Reply:
The precision values (size, horizontal precision, and vertical precision) are they are represented on-the-wire with a 4 bit mantissa and 4 bit exponent, limited to 9^9 according to the RFC. However, this encoding method can physically represent values up to 15^15, larger than what can fit in an int32. A 64 bit representation allows us to handle this case, despite the fact that it is undefined according to the RFC.

We use an int64 for parsing the altitude as it's an unsigned 32 bit number.  A number from 2^31 to 2^32 failes to be parsed by `Int32.of_string`. This is converted to an `int32` for as on the wire representation in `alt_parse`. Note that it requires conversion to an unsigned number when decoding in `alt_print`.
 
 The latitude and longitude store the degrees, minutes, and seconds of an arc. All fit well within an int32.
 
 So, all is fine for 32-bit systems.

Reply:
Oh - sorry - I just realized I may have misinterpreted your question. The reason why the precision values are stored as an `int` in `Loc.t` is because we only need 8 bits - 4 for the mantissa and 4 for the exponent. So all is find so long as the system is 8-bit or more :D